simulation_ws/src/test_nodes/nodes/docking_test.py

```diff
@@ -0,0 +1,152 @@
+            rospy.logerr("Error", e)
+            self.set_tag(name=self.test_name, value="Failed")
+            self.cancel_job()
+
+    def is_docked_cb(self, msg):
+        """
+        Cancel the test if "true" is published in /is_docked topic.
+        """
+        if self.is_docking is not True:
+            return
+        if msg.data is True and not self.is_docked and not self.is_cancelled:
+            rospy.loginfo("Docked")
+            self.is_docked = True
+            self.cancel_job()
+
+    def timeout_cb(self, msg):
+        """
+        Cancel the test if it times out.
+        The sim_timeout is based on the /clock topic (simulation time).
+        """
+        if self.is_docking is not True:
+            return
+        if (msg.clock.secs - self.starting_time_secs) > self.sim_timeout and not self.is_cancelled:
+            rospy.loginfo("Test timed out, cancelling job")
+            self.set_tag(name="Status", value="Failed")
+            self.set_tag(name="Timed_Out", value=str(True))
+            self.cancel_job()
+
+    def odom_cb(self, msg):
+        """
+        Cancel the test if robot is within threshold distance of the dock.
+        The distance is based on the odometry topic.
+        """
+        if self.is_docking is not True:
+            return
+        if self.distance_to_dock(msg) < self.threshold and not self.is_docked and not self.is_cancelled:
+            rospy.loginfo("Docked")
+            self.is_docked = True
+            self.cancel_job()
+        return
+
+    def cancel_job(self):
+        rospy.wait_for_service("/robomaker/job/cancel")
+        requestCancel = rospy.ServiceProxy("/robomaker/job/cancel", Cancel)
+        response = requestCancel()
+        if response.success:
+            self.is_cancelled = True
+            rospy.loginfo("Successfully requested cancel job")
+            self.set_tag(name=self.test_name + "_Time_Elapsed_End", value=str(time.time()).split(".",1)[0])
+        else:
+            rospy.logerr("Cancel request failed: %s", response.message)
+
+    def set_tag(self, name, value):
+        rospy.wait_for_service("/robomaker/job/add_tags")
+        requestAddTags = rospy.ServiceProxy("/robomaker/job/add_tags", AddTags)
+        tags = ([Tag(key=name, value=value)])
+        response = requestAddTags(tags)
+        if response.success:
+            rospy.loginfo("Successfully added tags: %s", tags)
+        else:
+            rospy.logerr(
+                "Add tags request failed for tags (%s): %s",
+                tags,
+                response.message)
+
+    def distance_to_dock(self, msg):
+        ros_tf = tf.TransformerROS()
+        p = PoseStamped()
+        p.pose.position.x = msg.pose.pose.position.x
+        p.pose.position.y = msg.pose.pose.position.y
+        p.header.frame_id = "odom"
+        p_in_map = ros_tf.transformPose("world", p)
+        return hypot(self.dock_x-p_in_map.pose.position.x, self.dock_y-p_in_map.pose.pose.position.y)
```

Comment:
```suggestion
        pose_in_map = ros_tf.transformPose("world", p)
        position_in_map = pose_in_map.pose.position
        return hypot(self.dock_x - position_in_map.x, self.dock_y - position_in_map.y)
```
Care of naming and format, please